<launch>

    <arg name="simulated" default="false" doc="Determines whether to instantiate real or simulated Arlobot"/>
    <arg name="enable_hal" default="true"/>
    <arg name="enable_sensor" default="true"/>

    <!--
    This launch file instantiates the arlobot nodes:
        - halnode   : provides an abstraction interface and communications with various devices, e.g., PSOC, IMU, etc
        - drivenode : implements the drive semantics for Arlobot accepting cmd_vel messages, publishing odometry
                      messages and communicating with the Arlobot hardware (via the HAL node), e.g., speed, distance,
                      orientation, etc.
        - sensornode: implements aggregation and transformation of various sensor inputs
    -->

    <group unless="$(arg simulated)" >
        <!-- Start the HAL node -->
        <node if="$(arg enable_hal)" name="halnode" pkg="arlobot_bringup" type="halnode.py" output="screen" respawn="true">
        </node>

        <!-- Start the sensor node -->
        <node if="$(arg enable_sensor)" name="sensornode" pkg="arlobot_bringup" type="sensornode.py" output="screen" respawn="true">
        </node>
    </group>

    <node name="drivenode" pkg="arlobot_bringup" type="drivenodefactory.py" output="screen" respawn="true">
    </node>

</launch>